pkg/sql/codegen/common.go

```diff
@@ -0,0 +1,129 @@
+			util.IntArrayToJSONString(c.FieldDesc.Shape)), nil
+	case *ir.BucketColumn:
+		sourceCode, err := GenerateFeatureColumnCode(c.SourceColumn, module)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf(
+			"%s.feature_column.bucketized_column(%s, boundaries=%s)",
+			module,
+			sourceCode,
+			util.IntArrayToJSONString(c.Boundaries)), nil
+	case *ir.CategoryIDColumn:
+		fm := c.GetFieldDesc()[0]
+		if len(fm.Vocabulary) > 0 {
+			vocabList := []string{}
+			for k := range fm.Vocabulary {
+				vocabList = append(vocabList, fmt.Sprintf("\"%s\"", k))
+			}
+			vocabCode := strings.Join(vocabList, ",")
+			return fmt.Sprintf("%s.feature_column.categorical_column_with_vocabulary_list(key=\"%s\", vocabulary_list=[%s])",
+				module, c.FieldDesc.Name, vocabCode), nil
+		}
+		return fmt.Sprintf("%s.feature_column.categorical_column_with_identity(key=\"%s\", num_buckets=%d)",
+			module, c.FieldDesc.Name, c.BucketSize), nil
+	case *ir.SeqCategoryIDColumn:
+		if !isTensorFlowModule(module) {
+			return "", fmt.Errorf("SEQ_CATEGORY_ID is only supported in TensorFlow models")
+		}
+		return fmt.Sprintf("%s.feature_column.sequence_categorical_column_with_identity(key=\"%s\", num_buckets=%d)",
+			module, c.FieldDesc.Name, c.BucketSize), nil
+	case *ir.CategoryHashColumn:
+		// FIXME(typhoonzero): do we need to support dtype other than int64?
+		dtype, err := toModuleDataType(c.FieldDesc.DType, module)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("%s.feature_column.categorical_column_with_hash_bucket(key=\"%s\", hash_bucket_size=%d, dtype=%s)",
+			module, c.FieldDesc.Name, c.BucketSize, dtype), nil
+	case *ir.CrossColumn:
+		var keysGenerated = make([]string, len(c.Keys))
+		for idx, key := range c.Keys {
+			if c, ok := key.(ir.FeatureColumn); ok {
+				code, err := GenerateFeatureColumnCode(c, module)
+				if err != nil {
+					return "", err
+				}
+				keysGenerated[idx] = code
+			} else {
+				return "", fmt.Errorf("field in cross column is not a FeatureColumn type: %v", key)
+			}
+		}
+		return fmt.Sprintf(
+			"%s.feature_column.crossed_column([%s], hash_bucket_size=%d)",
+			module, strings.Join(keysGenerated, ","), c.HashBucketSize), nil
+	case *ir.EmbeddingColumn:
+		if !isTensorFlowModule(module) {
+			return "", fmt.Errorf("EMBEDDING is only supported in TensorFlow models")
+		}
+
+		sourceCode, err := GenerateFeatureColumnCode(c.CategoryColumn, module)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("%s.feature_column.embedding_column(%s, dimension=%d, combiner=\"%s\")",
+			module, sourceCode, c.Dimension, c.Combiner), nil
+	case *ir.IndicatorColumn:
+		if !isTensorFlowModule(module) {
+			return "", fmt.Errorf("INDICATOR is only supported in TensorFlow models")
+		}
+
+		sourceCode, err := GenerateFeatureColumnCode(c.CategoryColumn, module)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("%s.feature_column.indicator_column(%s)", module, sourceCode), nil
+
```

Comment:
We don't need a blank line here. [Go switch exapmle](https://github.com/golang/go/wiki/Switch#switch)

Reply:
Done.